lib/collection/header.js

```diff
@@ -182,12 +195,20 @@ _.assign(Header, /** @lends Header */ {
      */
     unparse: function (headers, separator) {
         if (!_.isArray(headers) && !PropertyList.isPropertyList(headers)) {
-            return '';
+            return E;
         }
 
-        return headers.map(function (header) {
-            return header.key + ': ' + header.value;
-        }).join(separator ? separator : '\n');
+        return headers.map(Header.unparseSingle).join(separator ? separator : '\n');
+    },
+
+    /**
+     * Unparses a single Header.
+     * @param {String} header
+     * @returns {String}
+     */
+    unparseSingle: function (header) {
+        if (!_.isObject(header)) { return E; }
```

Comment:
The flow of the if condition can be reversed so that `return E;` is the fallback instead of the first check.